Adding POST APIs used to Upload customer TDE certificates in CMS.

nivimsft opened the pull request:
Commit contains appropriate tests

<!-- DO NOT DELETE THIS TEMPLATE -->

## Description
Adding POST APIs used to Upload customer TDE certificates in CMS code and tests
PR for spec changes https://github.com/Azure/azure-rest-api-specs/pull/3248

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [X] Please add REST spec PR link to the SDK PR
- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [X] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [X] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [X] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [X] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [X] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

nivimsft committed:
Adding POST APIs used to Upload customer TDE certificates in CMS.
Commit contains appropriate tests

dsgouda reviewed (commented):
Will approve once REST spec is merged and sdk regenerated.

jaredmoo commented on src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj:
```diff
@@ -13,6 +13,7 @@
 			<![CDATA[
 New features:
 - Added support for using sever level rule in vulnerability assessment baseline operations.
+- Enable customers to upload their TDE certificate to CMS
```
I published 1.17 today (on request from vulnerability assessment team), so now you need to bump this up to 1.18 and wipe out the old release notes (i.e. just line 15)

jaredmoo replied:
Bump to 1.18 both in this file and in AssemblyInfo.cs

nivimsft replied:
Sure will do

jaredmoo commented on src/SDKs/_metadata/sql_resource-manager.txt:
```diff
@@ -1,33 +1,33 @@
-GitHub fork: Azure
-Branch:      master
-Commit:      69592a62dd01225f16a9b398664c3b9d6ddc8268
+GitHub fork: nivimsft
```
You will need to generate from Azure/master before this PR can be merged

nivimsft replied:
Sounds good! Waiting for PR to be merged

jaredmoo commented on src/SDKs/SqlManagement/Sql.Tests/UploadTdeCertificateScenarioTests.cs:
```diff
@@ -0,0 +1,144 @@
+                // Update with values from a current MI on the region
+                //
+                string resourceGroup = ManagedInstanceResourceGroup;
+                string managedInstanceName = ManagedInstanceName;
```
Tests MUST be fully automated. This means that they must create all prerequisite resources during test setup. If this is not possible, the test must be skipped (see geo restore scenario test for example)

nivimsft committed:
Addressing comments from Jared:
- Updating version to 1.18 in .csproj and assemblyInfo
- regenerate client from Azure/Master,
- skipping tests with long setup time

dsgouda reviewed (approved):
LGTM

dsgouda commented:
Will merge when CIs pass

jaredmoo commented on src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj:
```diff
@@ -7,13 +7,13 @@
 New features:
-- Added support for using sever level rule in vulnerability assessment baseline operations.
 - Added support for using sever level Threat Detection.
+- Enable customers to upload their TDE certificate to CMS
```
Customers do not know what CMS is. This release note should be more like, "Added support for uploading TDE certificate into servers and managed instances."

nivimsft replied:
You are right, will address

jaredmoo reviewed (approved):
Looks good, release note needs a little edit.

nivimsft committed:
Revising release note

jaredmoo commented:
Thanks, now it's perfect!

nivimsft commented:
I like that, thanks for your help in making it so! :)
